### ISSUE TITLE
Changing volume via multimedia keys stalls game for 1s per volume increment.

### ISSUE THREAD

**Issue by toojays** [adventuregamestudio/ags#571]
This is on Ubuntu 18.04, with ags3 at commit 481c73a. I was testing with Gemini Rue, but do not believe this is game-specific.

My laptop has dedicated keys to increase and decrease the audio volume. Each time I press these keys, the game pauses for a second. This is annoying. I increase the volume, then have to wait a second to know if I should increase it some more, etc.

I believe the cause is that when I press these keys, the OS pops up a borderless window to indicate that the volume is increasing. That causes Allegro to call the ``display_switch_out_suspend()`` callback, which includes a one second sleep. There's no comment explaining why we need the sleep. Removing it seems to make life better for me.

```diff
--- a/Engine/ac/game.cpp
+++ b/Engine/ac/game.cpp
@@ -1974,8 +1974,6 @@ void display_switch_out_suspend()
         }
     }
 
-    rest(1000);
-
     // restore the callbacks
     SetMultitasking(0);
 ```

See https://youtu.be/7WjAmXHgVdg for a video of the problem, followed by what I experience after applying the above change.

**Comment by ivan-mogilko** [comment]
I don't know why it is there either. History shows that rest() call is there at least since the 3.2.1 SVN version.

**Comment by ivan-mogilko** [comment]
This may actually be general issue, because game does not pause strictly when you alt+tab, but instead runs for about 1 second more. I may be mistaken, but this "sleep" seem to be called in a separate thread, and game may continue running on background for some time contrary to player's expectation (this needs a double check).

PS. idk if this changes with SDL port.

**Comment by ivan-mogilko** [comment]
We have a SDL2-based version now, and this delay was removed. The problem has to be retested again (but likely it's gone now).

**Comment by ericoporto** [comment]
https://user-images.githubusercontent.com/2244442/116936590-7244e680-ac3e-11eb-992c-b11f34416e13.mp4

Good news: this issue is not a problem on master branch. (for some reason my capture software doesn't record the volume changing but it's changing!)

**Comment by ericoporto** [comment]
I need to rebuild with the specific commit @toojays specified and see if it does as I don't remember right now.

**Comment by ivan-mogilko** [comment]
> I need to rebuild with the specific commit @toojays specified and see if it does as I don't remember right now.

I think, it does not have to be exact commit, could be pre-sdl version, like latest 3.5.0 or 3.5.1, because this was never fixed afaik.

**Comment by ivan-mogilko** [comment]
@ericoporto have you had a chance to test this on release-3.5.1 or release-3.5.0?

**Comment by ivan-mogilko** [comment]
Closed, as quite likely fixed in 3.6.0. Please reopen or open a new report if something similar occurs in 3.6.0 or later.
